src/server/utils/inviteTokens.js

```diff
@@ -0,0 +1,47 @@
+import crypto from 'crypto';
+/**
+ * A secret token has the user id, an expiration, and a secret
+ * the expiration allows for invalidating on the client or server. No need to hit the DB with an expired token
+ * the user id allows for a quick db lookup in case you don't want to index on email (also eliminates pii)
+ * the secret keeps out attackers (proxy for rate limiting, IP blocking, still encouraged)
+ * storing it in the DB means there exists only 1, one-time use key, unlike a JWT, which has many, multi-use keys
+ */
+export function makeInviteToken(id, epocExpiration) {
+  return new Buffer(JSON.stringify({
+    id,
+    sec: crypto.randomBytes(8).toString('base64'),
```

Comment:
@mattkrick what were your thoughts with these random bytes? Were they to be used somehow?

The scheme I would lend myself toward is:

1. Create a secret key called `salt`, make it an environment variable
2. Generate a token containing only `id` and `exp`, stick it on an invite link
3. Compute `hash(tokenBuffer.toString('base64' + salt)` and stick it in the database
4. When the token needs to be validated, decode the JSON, recompute the hash, and check that it matches in the DB
5. If everything is good, process the invitation; else, error!